src/dataflow/src/source/source.proto

```diff
@@ -0,0 +1,26 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+// See https://developers.google.com/protocol-buffers for what's going on here.
+
+syntax = "proto3";
+
+package gen;
+
+message ProtoSourceTimestamp {
+    oneof partition_id {
+        int32 kafka = 1;
+        bool none = 2;
```

Comment:
super nit: `google.protobuf.Empty none = 2;` seems nicer but I don't know what implications it would have for encoded size and serde performance.

Reply:
Done